src/Compilers/Core/Portable/SourceGeneration/AdditionalSourcesCollection.cs

```diff
@@ -13,44 +14,108 @@ namespace Microsoft.CodeAnalysis
 {
     internal sealed class AdditionalSourcesCollection
```

Comment:
Is this a per generator collection?

Reply:
Yes, we create a new one for every invocation of every generator. 

---
In reply to: [435563892](https://github.com/dotnet/roslyn/pull/44768#discussion_r435563892) [](ancestors = 435563892)